state/docker_resource.go

```diff
@@ -109,10 +110,10 @@ func (dr DockerResource) Get(resourceID string) (io.ReadCloser, int64, error) {
 		return nil, -1, errors.Trace(err)
 	}
 	data, err := yaml.Marshal(
```

Comment:
I don't think it's the responsibility of the persistence layer to construct this as a stream here.
Get and Save should be consistent. If a higher layer needs a reader, it should be constructed there.